src/web/app/components/question-types/question-edit-answer-form/msq-question-edit-answer-form.component.spec.ts

```diff
@@ -23,7 +23,28 @@ describe('MsqQuestionEditAnswerFormComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create: should return true if compnent is successfully created', () => {
     expect(component).toBeTruthy();
   });
 });
+
+  it('updateNoneOfTheAbove: should update answers based on isNoneOfTheAboveEnabled', () => {
+    const component = new MsqQuestionEditAnswerFormComponent();
+    component.responseDetails = {
+      answers: ['A', 'B', 'C',]
+    };
+
+    // Simulate isNoneOfTheAboveEnabled being false
+    component.updateNoneOfTheAbove();
+
+    expect(component.responseDetails.answers).toEqual(['NoneOfTheAbove']);
```

Comment:
`responseDetails` has an `@Input()` decorator so it is set by the parent component, not `updateNoneOfTheAbove()`
`responseDetails` is not used for data binding in the html

so instead of testing the value of `responseDetails.answers`
test the behavior of `updateNoneOfTheAbove()`

it calls `this.triggerResponseDetailsChangeBatch()` with values for the properties `answers:` `isOther:` `otherFieldContent:`

create a spy for triggerResponseDetailsChangeBatch
https://jestjs.io/docs/jest-object#jestspyonobject-methodname

then use expect().toHaveBeenCalledWith();
https://jestjs.io/docs/expect#tohavebeencalledwitharg1-arg2-

you will need to look at the parent class of `MsqQuestionEditAnswerFormComponent`
which is `QuestionEditAnswerFormComponent`

it is in the file **question-edit-answer-form.ts**
it is in the same folder as your test